runtime/runtime-params-estimator/src/cases.rs

```diff
@@ -0,0 +1,371 @@
+    let medium_code = include_bytes!("../test-contract/res/medium_contract.wasm");
+    let large_code = include_bytes!("../test-contract/res/large_contract.wasm");
+    let curr_code = RefCell::new(small_code.to_vec());
+    let mut nonces: HashMap<usize, u64> = HashMap::new();
+    let mut accounts_deployed = HashSet::new();
+    let mut f = || {
+        let account_idx = loop {
+            let x = rand::thread_rng().gen::<usize>() % config.active_accounts;
+            if accounts_deployed.contains(&x) {
+                continue;
+            }
+            break x;
+        };
+        accounts_deployed.insert(account_idx);
+        let account_id = get_account_id(account_idx);
+        let signer = InMemorySigner::from_seed(&account_id, KeyType::ED25519, &account_id);
+        let nonce = *nonces.entry(account_idx).and_modify(|x| *x += 1).or_insert(1);
+        SignedTransaction::from_actions(
+            nonce as u64,
+            account_id.clone(),
+            account_id.clone(),
+            &signer,
+            vec![Action::DeployContract(DeployContractAction { code: curr_code.borrow().clone() })],
+            CryptoHash::default(),
+        )
+    };
+    let mut testbed =
+        measure_transactions("deploy", 1, Some(small_code.len()), &mut m, &config, None, &mut f);
+    *curr_code.borrow_mut() = medium_code.to_vec();
+    testbed = measure_transactions(
+        "deploy",
+        1,
+        Some(medium_code.len()),
+        &mut m,
+        &config,
+        Some(testbed),
+        &mut f,
+    );
+    *curr_code.borrow_mut() = large_code.to_vec();
+    testbed = measure_transactions(
+        "deploy",
+        1,
+        Some(large_code.len()),
+        &mut m,
+        &config,
+        Some(testbed),
+        &mut f,
+    );
+
+    let ad: Vec<_> = accounts_deployed.into_iter().collect();
+
+    // Measure the speed of processing function calls that do nothing.
+    let mut f = || {
+        let account_idx = *ad.as_slice().choose(&mut rand::thread_rng()).unwrap();
+        let account_id = get_account_id(account_idx);
+        let signer = InMemorySigner::from_seed(&account_id, KeyType::ED25519, &account_id);
+        let nonce = *nonces.entry(account_idx).and_modify(|x| *x += 1).or_insert(1);
+        let function_call = Action::FunctionCall(FunctionCallAction {
+            method_name: "noop".to_string(),
+            args: vec![],
+            gas: 10_000_000,
+            deposit: 0,
+        });
+        SignedTransaction::from_actions(
+            nonce as u64,
+            account_id.clone(),
+            account_id.clone(),
+            &signer,
+            vec![function_call],
+            CryptoHash::default(),
+        )
+    };
+    testbed = measure_transactions("call noop", 1, None, &mut m, &config, Some(testbed), &mut f);
+
+    // Measure the speed of all extern function calls.
+    testbed = measure_function("call_fixture10", 10, &mut m, testbed, &ad, &mut nonces, &config);
```

Comment:
... and these functions are measured 10/20/100 times. I mean how to figure out how many times makes sense to measure

Reply:
So these magic numbers have the following meaning:
* The number tells the given contract function how many times to run a certain operation in a loop;
* We want this operation to be executed as many time as possible to get statistical precision, however certain operations are too slow/expensive to run 100 times. So I figured out the constants that are large enough but do not make the code really slow.

In the future, we should have a more scientific approach for these constants (CC @icerove ), but for now we need to at least figure out the order of magnitude for fees costs, so I think these constants are good enough for now to give us the ballpark.

Reply:
Is this the case of

> Also, if create a transaction with multiple actions each of which is doing a function call the receipts produced by all function calls except the last one are going to be ingnored.

Reply:
No, I am not sure what similarity are you trying to point to. Here we create transactions where each transaction has one action and this action is a function call to the contract, but inside this contract there is a loop that repeats certain action several times. This is very different than having multiple actions (where each action is a function call) in one transactions.